gen/typescript/templates/client.ts.tmpl

```diff
@@ -12,7 +12,7 @@ export class {{.Name}} implements {{.Name | serviceInterfaceName}} {
 
   constructor(hostname: string, fetch: Fetch) {
     this.hostname = hostname
-    this.fetch = fetch
```

Comment:
This line wraps browser's `window.fetch()` with anonymous function.

As explained by Alex, the inner implementation of `window.fetch()` method calls self object (`this.`) in some browsers. In this case we replaced `this.` reference (the `window.` object) with an instance of the WebRPC client (`class {{.Name}}`). Thus, some browsers (Chrome/Firefox) were failing on different errors, ie `this.fetch() is not defined`.

This wrapper anonymous function hides `this.` object reference and thus fixes the issue.

"The wonders of JavaScript."